API/models/base.go

```diff
@@ -57,15 +58,39 @@ func init() {
 
 	client, err := mongo.NewClient(options.Client().ApplyURI(dbUri))
 	if err != nil {
-		log.Fatal(err)
 		println("Error while generating client")
+		log.Fatal(err)
 	}
 	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
 	err = client.Connect(ctx)
 	if err != nil {
-		log.Fatal(err)
 		println("Error while connecting")
+		log.Fatal(err)
+	}
+
+	//Check if API is authenticated
+	dbs, e := client.ListDatabaseNames(ctx, bson.D{}, options.ListDatabases())
+	if e != nil {
+		if strings.Contains(e.Error(), "requires authentication") {
+			log.Fatal("Authentication failed")
+		}
+		log.Fatal(e.Error())
+	} else {
+		found := false
+		for i := range dbs {
+			if dbs[i] == dbName {
+				found = true
+			}
+		}
```

Comment:
There is an unused CheckIfDBExists that seems to be doing that.

Reply:
I replaced it with the CheckIfDBExists function